yrs/src/block.rs

```diff
@@ -1334,7 +1337,10 @@ impl ItemContent {
         match self {
             ItemContent::Deleted(len) => encoder.write_len(*len - offset),
             ItemContent::Binary(buf) => encoder.write_buf(buf),
-            ItemContent::String(s) => encoder.write_string(&s.as_str()[(offset as usize)..]),
+            ItemContent::String(s) => {
+                let (_, right) = s.split_at(offset as usize, OffsetKind::Utf16);
```

Comment:
Fix 2: offsets are clock-relative, which means they are using UTF-16 encoding. We cannot just split string content by byte offset.